src/ocean/core/UnitTestRunner.d

```diff
@@ -58,8 +58,30 @@ import ocean.transition;
 
 import ocean.stdc.string: strdup, strlen, strncmp;
 import core.sys.posix.unistd: unlink;
-import core.sys.posix.libgen: basename;
-import core.sys.posix.sys.time: gettimeofday, timeval, timersub;
+
+import core.sys.posix.sys.time: gettimeofday, timeval;
+
+static if (__VERSION__ >= 2000 && __VERSION__ < 2073)
+{
+    extern(C) char* basename(char*);
+
+    extern (D): pure: @safe: @nogc: nothrow:
```

Comment:
You are right, it'll be applied to the whole module in the current state. I'll write it within a scope. And we actually won't need to define the D2 attributes here.

indeed `extern(D)` can be removed.